draft-ietf-taps-arch.md

```diff
@@ -442,7 +443,19 @@ If two different Protocol Stacks can be safely swapped, or raced in parallel (se
+3. Both stacks MUST offer the same security properties. The inclusion of transport security protocols {{I-D.ietf-taps-transport-security}} in a Protocol Stack adds additional restrictions to Protocol Stack equivalence. Security features and properties, such as cryptographic algorithms, peer authentication, and identity privacy vary across security protocols, and across versions of security protocols. Protocol equivalence ought not to be assumed for different protocols or protocol versions, even if they offer similar application configuration options. To ensure that security protocols are not incorrectly swapped, Transport Services systems SHOULD only automatically generate equivalent Protocol Stacks when the transport security protocols within the stacks are identical. Specifically, a transport system would consider protocols identical only if they are of the same type and version. For example, the same version of TLS running over two different transport protocol stacks are considered equivalent, whereas TLS 1.2 and TLS 1.3 {{RFC8446}} are not considered equivalent.
+
+### Separating Connection Groups {#groups}
+
+By default, all stored properties of the Implementation are shared within a process, such as cached protocol state, cached path state, and heuristics. This provides efficiency and convenience for the application, since the Transport System Implementation will be able to automatically optimize behavior.
+
+There are several reasons, however, that an application might want to isolate some Connections within a single process. These reasons include:
+
+- Privacy concerns about re-using cached protocol state that can lead to linkability. Sensitive state may include TLS session resumption state {{RFC8446}} and HTTP cookies {{RFC6265}}.
+- Privacy concerns about allowing Connections to multiplex together, which can tell a Remote Endpoint that all of the Connections are coming from the same application (for example, when Connections are multiplexed HTTP/2 or QUIC streams).
```

Comment:
This is a bit of a double edged sword since multiplexing hides information from the on-path observer. I might drop this one, since coalescing to the same origin is (IMO) fine.

Reply:
does the transport system implementation know enough when autocoalescing to ensure that it's the same origin, though? Thinking about cdn/cloud load balancers here...

Reply:
Yeah, I'd like to leave this. It's given as a concern an application may have; and when one is being paranoid (like for a private browsing case) it's nice to have a guarantee in usage like this.